### PR TITLE
Delete students from excel input

### PR DESCRIPTION
**Pull request by Aviral09**
- Added /delete_students route to admin
- Added a new del_students.html template
- Added an appropriate view to delete students from excel input

### CODE REVIEW

File: swd/main/views.py [hunk 1]
```diff
@@ -3436,3 +3436,86 @@ def upload_profile_pictures(request):
...
+                            username = 'f' + studentID[0:4] + studentID[8:12]
+
+                    else:
+                        if studentID[4] == 'P':
+                            username = 'p' + studentID[0:4] + studentID[8:11]
+                        elif studentID[4] == 'H':
+                            username = 'h' + studentID[0:4] + studentID[8:11]
+                        else:
+                            username = 'f' + studentID[0:4] + studentID[8:11]
+                    print(username)
```
Comment (khizirsiddiqui) [inline swd/main/views.py:3499]: You can remove this now.

File: swd/main/views.py [hunk 2]
```diff
@@ -3436,3 +3436,86 @@ def upload_profile_pictures(request):
...
+                    try:
+                        user = User.objects.get(username=username)
+                        user.delete()
+                        count = count + 1
+                        # student = Student.objects.get(bitsId=studentID)
+                        # student.delete()
+                    
+                    except User.DoesNotExist:
+                        # message_str += studentID + " failed"
+                        pass
```
Comment (khizirsiddiqui) [inline swd/main/views.py:3511]: The user should be notified in this case, pushing an error text to message framework can help.

File: swd/main/views.py [hunk 3]
```diff
@@ -3436,3 +3436,86 @@ def upload_profile_pictures(request):
...
+
+                    else:
+                        if studentID[4] == 'P':
+                            username = 'p' + studentID[0:4] + studentID[8:11]
+                        elif studentID[4] == 'H':
+                            username = 'h' + studentID[0:4] + studentID[8:11]
+                        else:
+                            username = 'f' + studentID[0:4] + studentID[8:11]
+                    print(username)
+                    password = User.objects.make_random_password()
```
Comment (khizirsiddiqui) [inline swd/main/views.py:3500]: Did you use this variable later? You can remove if it isn't needed.

File: swd/main/views.py [hunk 4]
```diff
@@ -3436,3 +3436,86 @@ def upload_profile_pictures(request):
                 request, "No folder selected. Please select at least one.")
        
     return render(request, "upload_profile_pictures.html", {})
+
+@user_passes_test(lambda u: u.is_superuser)
```
Comment (khizirsiddiqui) [inline swd/main/views.py:3440]: @ashadhaz are all the staff members superusers? If not, maybe we will need to change this user test in more functions.

File: swd/main/views.py (hunk 1 above)
Comment (Aviral09) [reply swd/main/views.py:3499]: Okay will remove print

File: swd/main/views.py (hunk 2 above)
Comment (Aviral09) [reply swd/main/views.py:3511]: Okay will add a message to the final message.

File: swd/main/views.py (hunk 3 above)
Comment (Aviral09) [reply swd/main/views.py:3500]: No, I'm not using password later. This will be removed.

File: swd/main/views.py (hunk 4 above)
Comment (ashadhaz) [reply swd/main/views.py:3440]: No, not all staff members are superusers, but only a superuser staff member is allowed to delete students, so this test will work just fine

**Review by ashadhaz — APPROVED** [review]

**Pull request merged**